cmd/genji/main.go

```diff
@@ -26,38 +25,91 @@ func main() {
 		},
 	}
 
+	app.Commands = []*cli.Command{
+		{
+			Name:      "insert",
+			Usage:     "Allows inserting documents from the command line",
+			UsageText: "genji insert [options] [arguments...]",
+			HideHelp:  true,
+			Flags: []cli.Flag{
+				&cli.StringFlag{
+					Name:     "engine",
+					Aliases:  []string{"e"},
+					Usage:    "Store engine bolt or badger",
+				},
+				&cli.StringFlag{
+					Name:     "db",
+					Usage:    "path of the database file",
+					Required: true,
+				},
+				&cli.StringFlag{
+					Name:     "table",
+					Aliases:  []string{"t"},
+					Usage:    "name of the table, it must already exist",
+					Required: true,
+				},
+			},
+			Action: func(c *cli.Context) error {
+				dbPath := fmt.Sprintf("%v", c.Value("db"))
+				table := fmt.Sprintf("%v", c.Value("table"))
+				// Use bolt as default engine.
+				engine := "bolt"
+				if c.IsSet("engine") {
+					engine = fmt.Sprintf("%v", c.Value("engine"))
+				}
+
+				data := c.Args().Slice()
+				err := shell.RunInsertCommand(engine, dbPath, table, data)
+				// Print error but return nil to avoid double error printing.
+				if err != nil {
+					_, _ = fmt.Fprintf(c.App.Writer, "error: %s\n", err.Error())
+					_ = cli.NewExitError(cli.ShowSubcommandHelp(c), 0)
+					return nil
+				}
+
+				return nil
+			},
+
+			OnUsageError: func(c *cli.Context, err error, isSubcommand bool) error {
+				_, _ = fmt.Fprintf(c.App.Writer, "error: %s\n", err.Error())
+				_ = cli.NewExitError(cli.ShowSubcommandHelp(c), 0)
+				// don't return err to avoid double err printing.
+				return nil
+			},
+		},
+	}
+
+	// Use for genji without commands
 	app.Action = func(c *cli.Context) error {
 		useBolt := c.Bool("bolt")
 		useBadger := c.Bool("badger")
 		if useBolt && useBadger {
 			return cli.NewExitError("cannot use bolt and badger options at the same time", 2)
 		}
 
-		dbpath := c.Args().First()
+		var opts *shell.Options
 
-		if (useBolt || useBadger) && dbpath == "" {
+		// set default engine.
+		opts = &shell.Options{Engine: "memory"}
+		opts.DBPath = c.Args().First()
+
+		if (useBolt || useBadger) && opts.DBPath == "" {
 			return cli.NewExitError("db path required when using bolt or badger", 2)
 		}
 
-		engine := "memory"
-
-		if useBolt || dbpath != "" {
-			engine = "bolt"
+		if useBolt || opts.DBPath != "" {
+			opts.Engine = "bolt"
 		}
 
 		if useBadger {
-			engine = "badger"
+			opts.Engine = "badger"
 		}
 
-		return shell.Run(&shell.Options{
-			Engine: engine,
-			DBPath: dbpath,
-		})
+		return shell.Run(opts)
 	}
 
-	err := app.Run(os.Args)
-	if err != nil {
-		log.Fatal(err)
+	if err := app.Run(os.Args); err != nil {
+		fmt.Println(err)
```

Comment:
Is there a reason we are printing the error this way?

Reply:
It was before using cli. I did it to delete the datetime.